docker/mxnet-model-server.sh

```diff
@@ -0,0 +1,163 @@
+#!/bin/bash
+
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+GUNICORN_ARGS='gunicorn'
+MXNET_ARGS='mxnet'
+NGINX_ARGS='nginx'
+MMS_ARG='mms'
+NGINX_CONFIG_FILE='/etc/nginx/conf.d/virtual.conf'
+MMS_CONFIG_FILE=''
+
+# Start MxNet model server. This function expects one argument which is the application config file.
+start_mms() 
+{
+    gunicorn_arguments=''
+    # TODO: Check if this process is already running . If yes ignore start and throw warning.
+
+    # This function expects one argument.
+    if [[ "$#" != 1 ]]
+    then
+        echo "ERROR: Wrong arguments given ($#) \"$@\""
+        exit 1
+    fi
+
+    MMS_CONFIG_FILE=$1
+    models=""
+    models_found=0
+
+    echo "INFO: Reading file $MMS_CONFIG_FILE"
+
+    # if the MMS CONFIG FILE is an empty string or if the file doesn't exist, throw error and exit
+    if [[ ( -z "${MMS_CONFIG_FILE##*( )}" ) || ( ! -f "$MMS_CONFIG_FILE" ) ]]
+    then
+        echo "ERROR: No configuration file given... $MMS_CONFIG_FILE"
+        exit 1
+    fi
+
+    while read -r line || [ -n "$line" ]
+    do
+        line="${line##*( )}"
+	    shopt -s nocasematch
+	    # Comments and empty lines should be ignored . # and $ are comment starters.
+	    if [[ ( -z "$line" ) || ( "$line" =~ ^\# ) || ( "$line" =~ ^\$ ) ]]
+     	then
+    	    continue
+	    fi
+	    # If a line starts with '[' treat it as a header
+	    if [[ "$line" =~ ^\[[A-Za-z\ ] ]]
+	    then
+    	    HEADER=$line
+	        if [[ ( "$HEADER" =~ "$NGINX_ARGS" ) ]] ; then
+	            rm -f $NGINX_CONFIG_FILE
+	            touch $NGINX_CONFIG_FILE
+	        fi
+	        continue
+	    fi
+
+	    if [[ "$HEADER" =~ "$GUNICORN_ARGS" ]] # Gunicorn args
+	    then
+    	    gunicorn_arguments="${gunicorn_arguments} $line "
+    	elif [[ "$HEADER" =~ "$MXNET_ARGS" ]] # MxNet Engine Args
+    	then
+    	    export $line
+    	elif [[ "$HEADER" =~ "$NGINX_ARGS" ]] # Nginx Args
+    	then
+    	    # MMS_HOST gets updated when you run Docker (run/exec) command with the environment variables MMS_HOST=$HOSTNAME
+  	        if [[ ( "$line" =~ "server_name" ) && ( ! -z ${MMS_HOST// } ) ]]
+   	        then
+   	            host_name=$MMS_HOST
+   	            line="server_name $host_name;"
```

Comment:
what's the value of setting nginx hostname in docker container? just logging?

Reply:
This will update the nginx configuration to use "$HOSTNAME" as the "server_name"